java/core/src/main/java/co/worklytics/psoxy/gateway/impl/CachingConfigServiceDecorator.java

```diff
@@ -53,33 +57,39 @@ public boolean supportsWriting() {
 
     @Override
     public void putConfigProperty(ConfigProperty property, String value) {
-        getCache().put(property, value);
+        if (!property.noCache()) {
+            getCache().put(property, value);
+        }
         delegate.putConfigProperty(property, value);
     }
 
     @SneakyThrows
     @Override
     public String getConfigPropertyOrError(ConfigProperty property) {
         return getConfigPropertyAsOptional(property)
-            .orElseThrow(() -> new Error("Psoxy misconfigured. Expected value for: " + property.name()));
+            .orElseThrow(() -> new NoSuchElementException("Psoxy misconfigured. Expected value for: " + property.name()));
     }
 
     @SneakyThrows
     @Override
     public Optional<String> getConfigPropertyAsOptional(ConfigProperty property) {
-        try {
-            String value = getCache().get(property);
-            if (Objects.equals(NEGATIVE_VALUE, value)) {
-                return Optional.empty();
-            } else {
-                return Optional.of(value);
-            }
-        } catch (ExecutionException e) {
-            //unwrap if possible, re-throw
-            if (e.getCause() == null) {
-                throw e;
-            } else {
-                throw e.getCause();
+        if (property.noCache()) {
+            return delegate.getConfigPropertyAsOptional(property);
+        } else {
+            try {
+                String value = getCache().get(property);
+                if (Objects.equals(NEGATIVE_VALUE, value)) {
+                    return Optional.empty();
+                } else {
+                    return Optional.of(value);
```

Comment:
```suggestion
                    return Optional.ofNullable(value);
```

Reply:
`value` should be `##NEGATIVE_VALUE##` if the value isn't in cache; I don't think it would be possible for `value` to ever be `null`.  I think you would have to configure it as `""` or something.